kythe/cxx/indexer/textproto/analyzer.cc

```diff
@@ -440,6 +440,70 @@ absl::Status TextprotoAnalyzer::AnalyzeEnumValue(const proto::VName& file_vname,
   return absl::OkStatus();
 }
 
+// TODO: refactor to share more code with AnalyzeEnumValue().
+absl::Status TextprotoAnalyzer::AnalyzeStringValue(
+    const proto::VName& file_vname, const Message& proto,
+    const FieldDescriptor& field, int start_offset) {
+  // Start after the last character of the field name.
+  re2::StringPiece input(textproto_content_.data(), textproto_content_.size());
+  input = input.substr(start_offset);
+
+  // Consume rest of field name, colon (optional).
+  ConsumeTextprotoWhitespace(&input);
+  if (!re2::RE2::Consume(&input, ":")) {
+    return absl::UnknownError("Failed to find ':' when analyzing string value");
+  }
+  ConsumeTextprotoWhitespace(&input);
+
+  const bool array_format =
+      field.is_repeated() && re2::RE2::Consume(&input, "\\[");
+  if (array_format) ConsumeTextprotoWhitespace(&input);
+
+  while (true) {
+    char c = input[0];
+    if (c != '"' && c != '\'') {
+      return absl::UnknownError("Can't find string");
+    }
+
+    // Use tokenizer to get range of entire string. The result contains the
+    // quotation marks.
+    std::string token_text;
+    {
+      // Note: doing input.ToString() for each field is *very* inefficient.
+      // TODO: use a ZeroCopyReader backed by a string_view, not a string, which
+      // requires copying.
+      std::istringstream iss(input.ToString());
+      google::protobuf::io::IstreamInputStream iis(&iss);
```

Comment:
`google::protobuf::io::ArrayInputStream iis(input.data(), input.size());` (will take care of the TODO)

Reply:
awesome, thanks!